src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php

```diff
@@ -0,0 +1,297 @@
+			$query_args['pattern-keywords'] = $keyword_id;
+		}
+
+		if ( $search_term ) {
+			$query_args['search'] = $search_term;
+		}
+
+		$api_url = add_query_arg(
+			array_map( 'rawurlencode', $query_args ),
+			'http://api.wordpress.org/patterns/1.0/'
+		);
+
+		if ( wp_http_supports( array( 'ssl' ) ) ) {
+			$api_url = set_url_scheme( $api_url, 'https' );
+		}
+
+		$wporg_response = wp_remote_get( $api_url );
+		$raw_patterns   = json_decode( wp_remote_retrieve_body( $wporg_response ) );
+
+		if ( is_wp_error( $wporg_response ) ) {
+			$wporg_response->add_data( array( 'status' => 500 ) );
+
+			return $wporg_response;
+		}
+
+		// Make sure w.org returned valid data.
+		if ( ! is_array( $raw_patterns ) ) {
+			return new WP_Error(
+				'pattern_api_failed',
+				sprintf(
+				/* translators: %s: Support forums URL. */
+					__( 'An unexpected error occurred. Something may be wrong with WordPress.org or this server&#8217;s configuration. If you continue to have problems, please try the <a href="%s">support forums</a>.' ),
+					__( 'https://wordpress.org/support/forums/' )
+				),
+				array(
+					'status'   => 500,
+					'response' => wp_remote_retrieve_body( $wporg_response ),
+				)
+			);
+		}
+
+		$response = array();
+
+		if ( $raw_patterns ) {
+			foreach ( $raw_patterns as $pattern ) {
+				$response[] = $this->prepare_response_for_collection(
+					$this->prepare_item_for_response( $pattern, $request )
+				);
+			}
+		}
+
+		return new WP_REST_Response( $response );
+	}
+
+	/**
+	 * Prepare a raw pattern before it's output in an API response.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param object          $raw_pattern A pattern from api.wordpress.org, before any changes.
+	 * @param WP_REST_Request $request     Request object.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function prepare_item_for_response( $raw_pattern, $request ) {
+		$prepared_pattern = array(
+			'id'             => absint( $raw_pattern->id ),
+			'title'          => sanitize_text_field( $raw_pattern->title->rendered ),
+			'content'        => wp_kses_post( $raw_pattern->pattern_content ),
+			'categories'     => array_map( 'sanitize_title', $raw_pattern->category_slugs ),
+			'keywords'       => array_map( 'sanitize_title', $raw_pattern->keyword_slugs ),
+			'description'    => sanitize_text_field( $raw_pattern->meta->wpop_description ),
+			'viewport_width' => absint( $raw_pattern->meta->wpop_viewport_width ),
+		);
+
+		$prepared_pattern = $this->add_additional_fields_to_object( $prepared_pattern, $request );
+
+		$response = new WP_REST_Response( $prepared_pattern );
+
+		/**
+		 * Filters the REST API response for a pattern.
+		 *
+		 * @since 5.8.0
+		 *
+		 * @param WP_REST_Response $response    The response object.
+		 * @param object           $raw_pattern The unprepared pattern.
+		 * @param WP_REST_Request  $request     The request object.
+		 */
+		return apply_filters( 'rest_prepare_application_password', $response, $raw_pattern, $request );
```

Comment:
Doh, it looks like I missed this copy/pasta in https://github.com/WordPress/gutenberg/pull/26578/

Reply:
@iandunn should it be removed or just changed? `rest_prepare_block_pattern`?

Reply:
Probably changed, `rest_prepare_block_pattern` sounds good to me.